### PR TITLE
Dev: Update radar data

### PR DESCRIPTION
**Pull request by Gritslav**
#### Description

The speed data that is transmitted from the radars has been updated. In the current version, one speed component was issued, which was calculated as the scalar product of the relative velocity vector of the hitted actor and the direction vector per actor. In the new version, the speed is divided into components along the axes Ox and Oy of the radar. The speed is considered relative to the radar.

#### Where has this been tested?

  * **Platform(s):** Windows 10
  * **Python version(s):** 3.8
  * **Unreal Engine version(s):** 4.26

#### Possible Drawbacks

Carla may no longer be supported by the current version of ros-bridge.

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/6313)

### CODE REVIEW

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp [hunk 1]
```diff
@@ -175,16 +182,27 @@ void ARadar::SendLineTraces(float DeltaTime)
...
   const FVector DeltaVelocity = (TargetVelocity - CurrentVelocity);
-  const float V = TO_METERS * FVector::DotProduct(DeltaVelocity, Direction);
+
+  FVector2D V(ActorTransform.TransformVector(DeltaVelocity));
+
+  V.X *= TO_METERS;
+  V.Y *= TO_METERS;
+
+
+  // Для дебага!
```
Comment (Blyron) [inline Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp:202]: Could you remove comments in other languages than English?

File: Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp [hunk 2]
```diff
@@ -175,16 +182,27 @@ void ARadar::SendLineTraces(float DeltaTime)
...
-  const float V = TO_METERS * FVector::DotProduct(DeltaVelocity, Direction);
+
+  FVector2D V(ActorTransform.TransformVector(DeltaVelocity));
+
+  V.X *= TO_METERS;
+  V.Y *= TO_METERS;
+
+
+  // Для дебага! 
+  //UE_LOG(LogTemp, Warning, TEXT("Vf: %s"), *DV.ToString());
```
Comment (Blyron) [inline Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp:203]: Could you delete commented code

**Comment by Blyron** [comment]
Hello! Could you resolve Radar.cpp conflicts and comments?

**Comment by Blyron** [comment]
There are two conflicts there still

**Comment by Gritslav** [comment]
> There are two conflicts there still

I think I fixed the conflicts, but I clicked the merge button without thinking. Do I understand correctly that nothing critical happened, and my changes were saved in the repository?

**Comment by Blyron** [comment]
Your changes are there still, could you update the backlog and then the PR would be ready to go in

**Comment by Gritslav** [comment]
Did I understand correctly that I need to make a description of my changes to the file CHANGELOG.md ? If so, how to arrange it correctly? Just add a line to the list of recent updates?

**Comment by Blyron** [comment]
Yeah, that's it add a line commenting this. Many people will read and maybe this will be affecting their code

**Comment by Blyron** [comment]
Hello,

We have been looking at your PR. We want to keep the previous version of velocity. You could add your velocity y and x on top of that. For now we are not merging your PR

**Pull request closed without merging**